provider/image.go

```diff
@@ -639,37 +645,39 @@ func marshalCachedImages(b resource.PropertyValue) ([]string, error) {
 	if !ok {
 		return cacheImages, fmt.Errorf("cacheFrom requires an `images` field")
 	}
-	if images.IsNull() {
+	if images.IsNull() || images.IsComputed() {
 		return cacheImages, nil
 	}
+
 	if !images.IsArray() {
 		return cacheImages, fmt.Errorf("the `images` field must be a list of strings")
 	}
 
 	stages := images.ArrayValue()
 	for _, img := range stages {
-		// if we are in preview, we cannot add an undefined Output so we skip to the next item
-		if img.IsNull() {
-			continue
+		if !img.IsNull() && !img.ContainsUnknowns() {
+			stage := img.StringValue()
+			cacheImages = append(cacheImages, stage)
 		}
-		stage := img.StringValue()
-		cacheImages = append(cacheImages, stage)
 	}
 	return cacheImages, nil
 }
 
 func marshalRegistry(r resource.PropertyValue) Registry {
 	var reg Registry
+
 	if !r.IsNull() {
-		if !r.ObjectValue()["server"].IsNull() {
+
+		if !r.ObjectValue()["server"].IsNull() && !r.ObjectValue()["server"].ContainsUnknowns() {
```

Comment:
If `r` isn't an object value or null, this function will panic. I'd be very nervous here vis a vis if `r` is unknown or secret.

I would recommend changing the type signature of this function to operate on a `resource.PropertyMap`. I'm commenting on something not in this PR, so feel free to leave that for later.